docs/tutorials/elasticsearch.ipynb

```diff
@@ -0,0 +1,750 @@
+        "\n",
+        "index_es_data(index=\"train\", es_data=train_es_data)\n",
+        "time.sleep(3)\n",
+        "index_es_data(index=\"test\", es_data=test_es_data)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2mOrfOYrHpQj"
+      },
+      "source": [
+        "## Prepare tfio datasets\n",
+        "\n",
+        "Once the data is available in the cluster, only `tensorflow-io` is required to stream the data from the indices. The `elasticsearch.ElasticsearchIODataset` class is utilized for this purpose. The class inherits from `tf.data.Dataset` and thus exposes all the useful functionalities of `tf.data.Dataset` out of the box.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "58q52py93jEf"
+      },
+      "source": [
+        "### Training dataset\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "HHOcitbW2_d1"
+      },
+      "outputs": [],
+      "source": [
+        "BATCH_SIZE=32\n",
+        "HEADERS = {\"Content-Type\": \"application/json\"}\n",
+        "\n",
+        "train_ds = tfio.experimental.elasticsearch.ElasticsearchIODataset(\n",
+        "        nodes=[ES_NODES],\n",
+        "        index=\"train\",\n",
+        "        doc_type=\"pet\",\n",
+        "        headers=HEADERS\n",
+        "    )\n",
+        "\n",
+        "# Prepare a tuple of (data, label)\n",
+        "train_ds = train_ds.map(lambda v: (v, v.pop(\"target\")))\n",
+        "train_ds = train_ds.batch(BATCH_SIZE)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Me0zgeCQIsKH"
+      },
+      "source": [
+        "### Testing dataset"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2R-I9hUgIcXR"
+      },
+      "outputs": [],
+      "source": [
+        "test_ds = tfio.experimental.elasticsearch.ElasticsearchIODataset(\n",
+        "        nodes=[ES_NODES],\n",
+        "        index=\"test\",\n",
+        "        doc_type=\"pet\",\n",
+        "        headers=HEADERS\n",
+        "    )\n",
+        "\n",
+        "# Prepare a tuple of (data, label)\n",
+        "test_ds = test_ds.map(lambda v: (v, v.pop(\"target\")))\n",
+        "test_ds = test_ds.batch(BATCH_SIZE)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IOD91R4BazMt"
+      },
+      "source": [
+        "Fetch a sample batch and observe the `TensorSpec`'s of the columns/key's. This will help in defining the keras pre-processing layers for training the `tf.keras` model.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-7DzTScGa6lW"
+      },
+      "outputs": [],
+      "source": [
+        "ds_iter = iter(train_ds)\n",
+        "batch = next(ds_iter)\n",
+        "batch"
```

Comment:
Big printouts like this are hard to read. Consider using:

```
features, label = next(ds_iter)
print(f"label: {label.numpy()[0]}")
{key: value.numpy()[0] for key,value in batch.items()}
```

Reply:
addressed in the commit